app/components/common/navbar/index.tsx

```diff
@@ -0,0 +1,228 @@
+              expand ? '' : 'justify-center'
+            } mb-0 md:mb-8 cursor-pointer  group`}
+            onClick={() => toggle(2)}
+          >
+            <div className="h-5 w-5 group-hover:stroke-slate-900 ">
+              <UsersIcon
+                className={`${
+                  myColor(2) ? ' stroke-slate-900' : ' stroke-stone-500'
+                }  group-hover:stroke-slate-900`}
+              />
+            </div>
+            <p
+              className={`block md:hidden text-xs  md:text-base  hover:text-slate-900 ${
+                myColor(0) ? ' text-slate-900' : ' text-stone-500 '
+              }`}
+            >
+              Events
+            </p>
+            {expand && (
+              <p
+                className={`pl-2 text-base hover:text-slate-900 ${
+                  myColor(2) ? ' text-slate-900' : ' text-stone-500'
+                }`}
+              >
+                Social Events
+              </p>
+            )}
+          </li>
+          <li
+            className={`flex flex-col md:flex-row md:hidden  items-center ${
+              expand ? '' : 'justify-center'
+            }  mb-0 md:mb-8 cursor-pointer group`}
+            onClick={() => toggle(3)}
+          >
+            <div className="h-5 w-5  group">
+              <Cog6ToothIcon
+                className={`${
+                  myColor(3) ? ' stroke-slate-900' : ' stroke-stone-500'
+                }  group-hover:stroke-slate-900`}
+              />
+            </div>
+
+            <p
+              className={` text-xs hover:text-slate-900 ${
+                myColor(3) ? ' text-slate-900' : ' text-stone-500'
+              }`}
+            >
+              Settings
+            </p>
+          </li>
+        </ul>
+      </div>
+      <div
+        className="md:basis-6/12"
+        onClick={() => {
+          setExpand((prev) => !prev);
+        }}
+      ></div>
+      <div className="hidden md:block md:basis-2/12">
+        <ul
+          className={`hidden md:flex flex-row md:flex-col ${
+            expand ? 'px-4 py-2' : 'p-0'
+          } mt-4 md:mt-0 md:text-sm md:font-medium md:border-0 bg-stone-50`}
+        >
+          <li
+            className={`flex items-center ${
+              expand ? '' : 'justify-center'
+            }  mb-0 md:mb-8 cursor-pointer group`}
+            onClick={() => toggle(3)}
+          >
+            <div className="h-5 w-5  group">
+              <Cog6ToothIcon
+                className={`${
+                  myColor(3) ? ' stroke-slate-900' : ' stroke-stone-500'
+                }  group-hover:stroke-slate-900`}
+              />
+            </div>
+            {expand && (
+              <p
+                className={`pl-2 text-base hover:text-slate-900 ${
+                  myColor(3) ? ' text-slate-900' : ' text-stone-500'
+                }`}
+              >
+                Settings
+              </p>
+            )}
+          </li>
```

Comment:
Instead of manually writing down all the links can we not make an array of objects where each object represents an individual link? 

An example of the would be: 

`
[
  {
      name: "home",
      href: "/dashboard", 
      icon: Home
  },
  .....
]
`

Reply:
yes , i have implemented the changes